src/main/java/tech/sourced/siva/SivaUnpacker.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * @see SivaReader#getEntry(IndexEntry)
+ */
+public class SivaUnpacker {
+    private static final Logger log = LoggerFactory.getLogger(SivaUnpacker.class);
+    private final SivaReader reader;
+    private Boolean ignorePerms = true;
+
+    public SivaUnpacker(SivaReader reader) {
+        this.reader = reader;
+    }
+
+    public SivaUnpacker(SivaReader reader, Boolean ignorePermissions) {
+        this(reader);
+        this.ignorePerms = ignorePermissions;
+    }
+
+    public SivaUnpacker(String sivaFilePath) throws SivaException {
+        try {
+            this.reader = new SivaReader(new File(sivaFilePath));
+        } catch (FileNotFoundException e) {
+            throw new SivaException("Could not read " + sivaFilePath, e);
+        }
+    }
+
+    public SivaUnpacker(String sivaFilePath, Boolean ignorePermissions) throws SivaException {
+        this(sivaFilePath);
+        this.ignorePerms = ignorePermissions;
+    }
+
+    public void unpack(String dstDir) throws SivaException {
+        File dst = new File(dstDir);
+        if (!dst.exists()) {
+            log.info("Directory " + dstDir + " does not exist, creating " + dst.getAbsolutePath());
+            if (!dst.mkdirs()) {
+                throw new SivaException("Failed to created a directory: " + dst.getAbsolutePath());
+            }
+        }
+
+        List<IndexEntry> entries = this.reader.getIndex().getFilteredIndex().getEntries();
+        for (IndexEntry entry : entries) {
+            this.extractEntry(entry, dstDir);
+        }
+        this.reader.close();
+    }
+
+    private void extractEntry(IndexEntry entry, String dstDir) throws SivaException {
+        this.extractEntry(entry, dstDir, null);
+    }
+
+    private void extractEntry(IndexEntry entry, String dstDir, CRC32 crc32) throws SivaException {
+        File dstFile = Paths.get(dstDir, entry.getName()).toAbsolutePath().toFile();
+
+        InputStream src = this.reader.getEntry(entry);
+        OutputStream dst = null;
+        try {
+            dst = createFileFor(dstFile);
+            IOUtils.copy(src, dst);
+
+            if (!this.ignorePerms) {
+                Set<PosixFilePermission> perms = entry.getFileMode();
+                Files.setPosixFilePermissions(dstFile.toPath(), perms);
+            }
+        } catch (FileNotFoundException e) {
+            log.error("Failed to create " + entry.getName() + " in " + dstDir, e);
+            return;
+        } catch (IOException e) {
+            log.error("Failed to copy " + entry.getName() + " to " + dstDir, e);
+            return;
+        } finally {
+            close(src);
+            close(dst);
+        }
+
+        if (crc32 != null) {
+            calculateCRC(entry, crc32, src);
+        }
+    }
+
+    private OutputStream createFileFor(File dst) throws FileNotFoundException {
+        if (dst.exists()) { // warn
+            log.info("File " + dst + " already exist, content will be overwritten");
```

Comment:
If the content already exists, you should do nothing.

Reply:
Current implementation overwrites the content by default. 

It can be updated in subsequent PRs, if desired.

Reply:
change it please, overwrite files by default is not an expected behavior at all.